### PR TITLE
Fixes #31422 - Update Rubocop to new Lint/MissingSuper

### PR DESCRIPTION
**Pull request by chris1984**

### CODE REVIEW

**Comment by chris1984** [comment]
@jlsherrill here are the remaining lint failures with fixing things up in the test folder.

**Comment by jlsherrill** [comment]
looks like that change actually broke some tests :(    I think _() conflicts with the i18n, and thus we'd have to use _{}

Maybe just disable that cop for now?

File: app/lib/actions/katello/host/hypervisors_update.rb [hunk 1]
```diff
@@ -54,10 +54,10 @@ def load_hosts_guests
 
         # Loads all resources needed for refreshing subscription facet
         def load_resources
-          @organizations = ::Organization.where(label: hypervisors_field(:organization_label)).map { |org| [org.label, org] }.to_h
+          @organizations = ::Organization.where(label: hypervisors_field(:organization_label)).index_by { |org| org.label }
 
           candlepin_data = ::Katello::Resources::Candlepin::Consumer.get_all_with_facts(hypervisors_field(:uuid))
-          @candlepin_attributes = candlepin_data.map { |consumer| [consumer[:uuid], consumer] }.to_h
+          @candlepin_attributes = candlepin_data.index_by { |consumer| [consumer[:uuid], consumer] }
```
Comment (jlsherrill) [inline app/lib/actions/katello/host/hypervisors_update.rb:60]: this should be    ```.index_by { |consumer| consumer[:uuid] }```

File: app/lib/actions/katello/host/hypervisors_update.rb [hunk 2]
```diff
@@ -70,7 +70,7 @@ def load_hosts_by_uuid
           hosts_by_uuid.each do |host|
             validate_host_organization(host, host.organization.try(:id))
           end
-          hosts_by_uuid.map { |host| [host.subscription_facet.uuid, host] }.to_h
+          hosts_by_uuid.index_by { |host| [host.subscription_facet.uuid, host] }
```
Comment (jlsherrill) [inline app/lib/actions/katello/host/hypervisors_update.rb:73]:
```
.index_by { |host| host.subscription_facet.uuid }
```

File: app/lib/actions/katello/host/hypervisors_update.rb [hunk 3]
```diff
@@ -82,7 +82,7 @@ def load_hosts_by_duplicate_name
             validate_host_organization(host, duplicate_name_orgs[host.name].try(:id))
           end
 
-          hosts_by_dup_name.map { |host| [duplicate_names[host.name], host] }.to_h
+          hosts_by_dup_name.index_by { |host| [duplicate_names[host.name], host] }
```
Comment (jlsherrill) [inline app/lib/actions/katello/host/hypervisors_update.rb:85]: ```.index_by { |host| duplicate_names[host.name] }```

**Comment by jlsherrill** [comment]
few comments, but everything else looks good :+1:

File: app/lib/actions/katello/host/hypervisors_update.rb (hunk 1 above)
Comment (chris1984) [reply app/lib/actions/katello/host/hypervisors_update.rb]: Fixed

File: app/lib/actions/katello/host/hypervisors_update.rb (hunk 2 above)
Comment (chris1984) [reply app/lib/actions/katello/host/hypervisors_update.rb]: Fixed

File: app/lib/actions/katello/host/hypervisors_update.rb (hunk 3 above)
Comment (chris1984) [reply app/lib/actions/katello/host/hypervisors_update.rb]: Fixed

**Comment by chris1984** [comment]
@jlsherrill all green :)

**Comment by jlsherrill** [comment]
yay!!! Can you open an issue to re-enable those 4 cops?

**Review by jlsherrill — APPROVED** [review]

**Comment by chris1984** [comment]
> yay!!! Can you open an issue to re-enable those 4 cops?

Opened this guy:

https://projects.theforeman.org/issues/31436

**Pull request merged**